### PR TITLE
Rouding error fixed (#488)

### PR DESCRIPTION
**Pull request by jas-valgotar**
- Issue was caused by floating point error due to multiplication. 
- Changed multiplication of "m" to division. Also added new test cases!

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 1]
```diff
@@ -689,17 +689,17 @@ internal static double Round(double number, double digits, RoundType rt = RoundT
                 return number;
             }
 
-            var m = Math.Pow(10d, -dg);
-            var eps = m / 1e12d; // used to manage rounding of 1.4499999999999999999996
+            var m = Math.Pow(10d, dg);
+            var eps = 1 / (m * 1e12d); // used to manage rounding of 1.4499999999999999999996
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:693]: This is great. It's a subtle fix - can you add a comment to explain why it works? (that also acts to warn future developers from accidentally undoing it)

**Review by MikeStall — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: will do!

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 2]
```diff
@@ -689,6 +689,7 @@ internal static double Round(double number, double digits, RoundType rt = RoundT
                 return number;
             }
 
+            // Dividing by m, since multiplication was introducing floating point error
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:692]: Maybe it's a github bug, but it looks like you removed the actual fix in the latest iteration?

Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:692]: Yes, I noticed that. Fixing that!  Thanks!

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**